include/tvm/tir/function.h

```diff
@@ -90,13 +90,16 @@ class PrimFuncNode : public BaseFuncNode {
    *  will make program analysis much easier.
    */
   Map<tir::Var, Buffer> buffer_map;
+  /*! \brief The resource handle to be used by the function when accessing platform resources */
+  tir::Var resource_handle;
```

Comment:
it seems like this should be a property of the call site rather than a function. how come you want to add it here?

Reply:
This is referring to the variable used in the function to pass into further calls, similar to the params above except we don't treat it as a parameter which would get packed.

Reply:
hmm, but is there just one resource_handle associated with each function? suppose you had two instances of the same accelerator and wanted to launch the same compute function twice concurrently? also, what about the AOT top-level function, which should have a struct like `TVMDevices_my_model`?